### PR TITLE
libreswan: add package and service to nixos

### PR DESCRIPTION
**Pull request by afranchuk**
###### Things done:
- [x] Tested via `nix.useChroot`.
- [x] Built on platform(s): linux.
- [x] Tested compilation of all pkgs that depend on this change. (none)
- [x] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I have tested the resulting binaries/services and have been able to use the package to successfully connect to a rather temperamental VPN (to which strongswan couldn't connect).

### CODE REVIEW

File: nixos/modules/services/networking/libreswan.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.libreswan.enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/networking/libreswan.nix]: Please use `mkEnableOption`.

File: pkgs/tools/networking/libreswan/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  # Hack to make install work
+  installFlags = "FINALVARDIR=\${out}/var FINALSYSCONFDIR=\${out}/etc";
+
+  postInstall = ''
+    #ln -s $out/lib/systemd/system/ipsec.service $out/etc/systemd/system/ipsec.service
+  '';
+
+  meta = {
+    homepage = "https://libreswan.org";
+    description = "A free software implementation of the most widely supported and standarized VPN protocol based on ('IPsec') and the Internet Key Exchange ('IKE').";
```
Comment (jagajaga) [inline pkgs/tools/networking/libreswan/default.nix]: Please make description by our rules https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

**Comment by afranchuk** [comment]
The Travis CI seems to have failed somewhere unrelated to this pull request.

**Comment by jagajaga** [comment]
Please squash commits into one.

**Comment by abbradar** [comment]
I've reviewed the package itself and also noticed `environment` now. This should be the last round I think -- please bear with me and sorry for nitpicking!

**Comment by afranchuk** [comment]
I need to test the latest commit tomorrow just to make sure everything still works. I'm not with my NixOS system right now.

**Comment by abbradar** [comment]
Phew, I think that was the last round of remarks ~_^. Waiting for your testing and merging then. Thank you very much for your work and for your responsiveness!

**Comment by afranchuk** [comment]
Thank you for all of your help!

I just tested and made a few changes to fix things, it's good to go now!

**Comment by afranchuk** [comment]
Whoops. I didn't realize the build was auto-detecting the init system. It should be fixed now. (I used `nix.useChroot` to verify).

**Comment by abbradar** [comment]
Have you tried `enableParallelBuilding`? I ask because I got lazy to wait till the build finishes and enabled this for me locally -- it looks fine, so unless you have tested it and got race conditions I can merge it with the option enabled.

**Comment by afranchuk** [comment]
Actually (and I forget exactly where I read this) I think there is some concern in the libreswan build process with regards to parallel building. I only remember there was a specific reason I omitted it... So for now leave that out.

**Comment by abbradar** [comment]
I've explicitly specified that we don't support it in https://github.com/NixOS/nixpkgs/commit/f442830c55517c73a5e320d2bd892776db4216df. Thanks for your work!

**Comment by afranchuk** [comment]
Awesome. Thanks for your help!

**Pull request merged**